components/cover-image.js

```diff
@@ -0,0 +1,17 @@
+import Link from "next/link";
+
+export default function CoverImage({ title, url, slug }) {
+  return (
+    <div className="sm:mx-0">
+      {slug ? (
+        <Link as={`/posts/${slug}`} href="/posts/[slug]">
```

Comment:
With the latest version of Next.js, `as` is no longer needed!
```suggestion
        <Link href={`/posts/${slug}`}>
```